hibernate-cache/src/test/java/org/infinispan/test/hibernate/cache/functional/AbstractNonInvalidationTest.java

```diff
@@ -128,11 +130,11 @@ public void cleanup() throws Exception {
                awaitOrThrow(preFlushLatch);
             }
             s.flush();
-         } catch (OptimisticLockException e) {
+         } catch (StaleStateException e) {
```

Comment:
please use `OptimisticLockException | StaleStateException e`

Reply:
I'm not sure about this. 

`StateStateException` is what's expected in Hibernate 5.1, and `OptimisticLockException` in Hibernate 5.2. To use `OptimisticLockException | StaleStateException e` here, you'd be implying that in Hibernate 5.1, either can be thrown, which is not strictly true...

Your suggestion is convenient in the sense that it allows you easily deal with either Hibernate version though. Not fully convinced by it though.